pkgs/applications/editors/emacs/elisp-packages/manual-packages/color-theme-solarized/default.nix

```diff
@@ -1,26 +1,41 @@
 { lib
-, trivialBuild
-, fetchFromGitHub
 , color-theme
+, fetchFromGitHub
+, melpaBuild
+, writeText
 }:
 
-trivialBuild {
-  pname = "color-theme-solarized";
-  version = "0.pre+unstable=2017-10-24";
+let
+  argset = {
+    pname = "color-theme-solarized";
+    ename = "color-theme-solarized";
+    version = "20230209.837";
 
-  src = fetchFromGitHub {
-    owner = "sellout";
-    repo = "emacs-color-theme-solarized";
-    rev = "f3ca8902ea056fb8e46cb09f09c96294e31cd4ee";
-    hash = "sha256-oxX0lo6sxotEiR3nPrKPE9H01HKB3ohB/p8eEHFTp5k=";
-  };
+    src = fetchFromGitHub {
+      owner = "sellout";
+      repo = "emacs-color-theme-solarized";
+      rev = "b186e5d62d0b83cbf5cf38f7eb7a199dea9a3ee3";
+      hash = "sha256-7E8r56dzfD06tsQEnqU5mWSbwz9x9QPbzken2J/fhlg=";
+    };
+
+    packageRequires = [
+      color-theme
+    ];
+
+    commit = argset.src.rev;
 
-  packageRequires = [ color-theme ];
+    recipe = writeText "recipe" ''
+      (color-theme-solarized
+       :repo "sellout/emacs-color-theme-solarized"
+       :fetcher github)
+    '';
 
-  meta = with lib; {
-    homepage = "http://ethanschoonover.com/solarized";
-    description = "Precision colors for machines and people; Emacs implementation";
-    license = licenses.mit;
-    maintainers = with maintainers; [ samuelrivas AndersonTorres ];
+    meta =  {
+      homepage = "http://ethanschoonover.com/solarized";
```

Comment:
This is not the home page for this package.  It's the home page for the Solarized theme.

And please use https.

Reply:
THe HTTP**S** link fails sometimes. But OK, I will quote the GitHub link.